src/Data/List/ZipList.purs

```diff
@@ -25,6 +28,24 @@ newtype ZipList a = ZipList (List a)
 runZipList :: forall a. ZipList a -> List a
 runZipList (ZipList xs) = xs
 
+-- | Transpose any structure which is both `Foldable` and `Unfoldable`,
+-- | via `ZipList`; note that `sequence` for `ZipList` is the same as
+-- | `transpose`.
+transpose :: forall f a. (Foldable f, Unfoldable f) => f (f a) -> f (f a)
+transpose =
```

Comment:
My point is just that there is a difference between

- a function which is initially written in terms of `List` and then generalized because `Foldable` is basically just `toList`, and
- a function which is by definition, already generalized to work with some type class.

`Foldable` is a particularly confusing case since it basically defines a monoid homomorphism to `List`. But in the first case, a user can always choose to easily generalize to `Foldable` using a call to `toList`.